rules/TypeDeclaration/TypeNormalizer.php

```diff
@@ -87,19 +85,6 @@ public function normalizeArrayTypeAndArrayNever(Type $type): Type
                 return $traversedType;
             }
 
-            if ($traversedType instanceof UnionType) {
-                $traversedTypeTypes = $traversedType->getTypes();
-                $countTraversedTypes = count($traversedTypeTypes);
-
-                $collectedTypes = $this->getCollectedTypes($traversedTypeTypes);
-                $countCollectedTypes = count($collectedTypes);
-
-                // re-create new union types
-                if ($countTraversedTypes !== $countCollectedTypes && $countTraversedTypes > 2) {
-                    return $this->typeFactory->createMixedPassedOrUnionType($collectedTypes);
-                }
-            }
```

Comment:
this was used by `ReturnTypeDeclarationRector` which read docblock https://github.com/rectorphp/rector-src/commit/20153a915bbbbd665084f44f725a10fc8c6b778f

which now we use native type for return type inferer so it should safe to remove